lib/src/soloud.dart

```diff
@@ -628,6 +629,46 @@ interface class SoLoud {
     return completer.future;
   }
 
+  /// Stops the engine and disposes of all resources, including sounds
+  /// and the audio isolate in an synchronous way.
+  /// 
+  /// This method is meant to be called when exiting the app. For example
+  /// within the `dispose()` of the uppermost widget in the tree
+  /// or inside [AppLifecycleListener.onExitRequested].
+  /// 
+  /// During the normal app life cycle and if you want to shutdown the player,
+  /// please use [shutdown] which safer and it is meant to throw errors.
+  void deinit() {
+    _log.finest('deinit() called');
+    /// check if we are in the middle of an initialization.
+    if (_initializeCompleter != null) {
+      _log.warning('deinit() called while already initializing.');
+      _initializeCompleter = null;
+    }
+
+    /// if not already initialized, just return.
+    if (!_isInitialized) {
+      // The engine isn't initialized.
+      _log.warning('deinit() called when the engine is not initialized');
+      return;
+    }
+
+    /// reset broadcast and kill isolate
+    _isolateToMainStream?.close();
+    _isolateToMainStream = null;
+    _mainToIsolateStream = null;
+    _returnedEvent?.close();
+    _returnedEvent = null;
+    _isolate?.kill();
+    _isolate = null;
+    _isEngineInitialized = false;
+    _isInitialized = false;
+
+    SoLoudController().soLoudFFI.disposeAllSound();
+    SoLoudController().soLoudFFI.dispose();
```

Comment:
Just to be clear, this actually calls [`deinit()` on the C++ side](https://solhsa.com/soloud/basics.html), correct?

Reply:
Correct! An FFI method was missing in `bindings_player_ffi.dart` to check this. I double-checked that the player is really disposed even after 0ms since the end of `initialized()`. So we can think about the [previous comment](https://github.com/alnitak/flutter_soloud/pull/56#issuecomment-2003774773).